libraries/standard/mqtt/src/mqtt.c

```diff
@@ -1056,13 +1060,15 @@ static MQTTStatus_t sendPublish( MQTTContext_t * pContext,
     assert( pContext != NULL );
     assert( pPublishInfo != NULL );
     assert( headerSize > 0 );
+    assert( pContext->networkBuffer.pBuffer != NULL );
+    assert( !( pPublishInfo->payloadLength > 0 ) || ( pPublishInfo->pPayload != NULL ) );
 
     /* Send header first. */
     bytesSent = sendPacket( pContext,
                             pContext->networkBuffer.pBuffer,
                             headerSize );
 
-    if( ( bytesSent < 0 ) || ( ( size_t ) bytesSent != headerSize ) )
+    if( bytesSent < 0 )
```

Comment:
This was deleted because with the assert on too many bytes, sendPacket will return only a negative network error code OR the number of bytes requested to send.